pkg/cmd/server/origin/legacy.go

```diff
@@ -206,7 +207,8 @@ func LegacyStorage(storage map[schema.GroupVersion]map[string]rest.Storage) map[
 				case "deploymentConfigs":
 					restStorage := s.(*deploymentconfigetcd.REST)
 					store := *restStorage.Store
-					restStorage.DeleteStrategy = orphanByDefault(store.DeleteStrategy)
```

Comment:
@tnozicka that is right, it should be `store` and not `restStorage` (@deads2k).

I think we should fix this bug and backport to 3.5, @deads2k @smarterclayton thoughts?